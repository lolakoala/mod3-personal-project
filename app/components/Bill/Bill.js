import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Bill extends Component {
  markBillPaid = (billId, userId, usersHouse) => {
    this.props.markBillPaid(billId, userId, usersHouse);
    this.forceUpdate();
  }

  unPaid = id => {
    const { currentUser } = this.props;
    return id === currentUser.id ? 'Mark as Paid' : 'Not Yet Paid';
  }

  render() {
    const { usersHouse } = this.props;
    const bill = usersHouse.bills.find(bill => bill.id === this.props.match.params.id);
    return (
      <div id='bill-page'>
        <p className='title'>{`Bill: ${bill.title}`}</p>
        <div className='due-total'>
          <p className='duedate'>{`Due: ${bill.duedate}`}</p>
          <p className='total'>{`Total: ${bill.total}`}</p>
        </div>
        <p className='details'>{`Details: ${bill.details}`}</p>
        <p className='split'>House Split...</p>
        {bill.allUsersTotals.map(user => {
          const name = usersHouse.users.find(houseMember => houseMember.id === user.id).name;
          return (<div key={`${bill.parsedDuedate} ${user.id}`} className='user-split'>
            <p>{`${name}: ${user.total}`}</p>
            <div className='mark-paid' onClick={() => this.markBillPaid(bill.id, user.id, usersHouse)}><p>{user.paid ? 'Paid' : this.unPaid(user.id)}</p></div>
          </div>);
        })}
        <p className='posted-by-on'>{`Posted by ${bill.postedBy.name} on ${bill.datePosted}`}</p>
      </div>
    );

  }
}

export default Bill;

Bill.propTypes = {
  usersHouse: PropTypes.object,
  markBillPaid: PropTypes.func,
  match: PropTypes.object,
  currentUser: PropTypes.object
};
