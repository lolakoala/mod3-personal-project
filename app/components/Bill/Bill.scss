#bill-page {
  margin: 20px;
  height: auto;
  width: 280px;
  border-radius: 5px;
  background-color: rgba(141, 132, 73, 0.7);
  color: #3d6aa2;
  padding: 20px;
  font-size: 20px;
}

.user-split {
  height: auto;
  width: 90%;
  background-color: #F4CCE2;
  border-radius: 5px;
  margin-top: 10px;
  display: flex;
  padding: 5px;
  position: relative;
}

.user-split .mark-paid {
  height: 100%;
  width: 85px;
  background-color: #eb6b3c;
  position: absolute;
  top: 0;
  right: 0;
  color: #F4CCE2;
  display: inline-block;
  font-size: 20px;
  text-align: center;
  border-radius: 5px;
  // padding-top: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.mark-paid:hover {
  cursor: pointer;
  background-color: #F4CCE2;
  color: #eb6b3c;
}

#bill-page .title {
  height: auto;
  width: 90%;
  background-color: #F4CCE2;
  border-radius: 5px;
  padding: 5px;
  text-align: center;
  font-size: 30px;
  margin-top: 10px;
}

#bill-page .posted-by-on {
  display: inline-block;
  margin-top: 10px;
}

#bill-page .due-total {
  height: auto;
  width: 90%;
  background-color: #F4CCE2;
  border-radius: 5px;
  padding: 5px;
  text-align: center;
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
}

#bill-page .details {
  height: auto;
  width: 90%;
  background-color: #F4CCE2;
  border-radius: 5px;
  padding: 5px;
  margin-top: 10px;
  margin-bottom: 10px;
}
