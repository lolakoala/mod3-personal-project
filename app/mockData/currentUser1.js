const currentUser1 = {
  id : "RVde1m9BhkQBLZlTJNsFlQZXqP43",
  name: "Amanda Brenner",
  house: "Dog9"
};

export default currentUser1;
