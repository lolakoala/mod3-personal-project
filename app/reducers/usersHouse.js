import firebase from '../firebase.js';

const usersHouse = (state = {}, action) => {
  const housesRef = firebase.database().ref('houses');
  switch (action.type) {
  case 'CREATE_HOUSE':
    housesRef.push(action.house);
    return action.house;
  case 'GET_HOUSE':
    const updatedHouse = Object.assign(action.usersHouse, { users: [...action.usersHouse.users, action.user] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updatedHouse);
    return updatedHouse;
  case 'LOGIN_SUCCESS':
    return action.usersHouse;
  case 'SIGNOUT':
    return {};
  case 'LEAVE_HOUSE':
    const deleteUser = action.usersHouse.users.filter(user => user.id !== action.currentUser.id);
    const houseWithoutUser = Object.assign(action.usersHouse, { users: deleteUser });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithoutUser);
    return {};
  case 'ADD_BILL':
    const takeOutFakeBills = action.usersHouse.bills.filter(bill => bill.title !== 'fake');
    const houseWithBill = Object.assign(action.usersHouse, { bills: [...takeOutFakeBills, action.bill] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithBill);
    return houseWithBill;
  case 'ADD_BULLETIN':
    const takeOutFakeBulletins = action.usersHouse.bulletins.filter(bulletin => bulletin.title !== 'fake');
    const houseWithBulletin = Object.assign(action.usersHouse, { bulletins: [...takeOutFakeBulletins, action.bulletin] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithBulletin);
    return houseWithBulletin;
  case 'ADD_CHORE':
    const takeOutFakeChore = action.usersHouse.chores.filter(chore => chore.title !== 'fake');
    const houseWithChore = Object.assign(action.usersHouse, { chores: [...takeOutFakeChore, action.chore] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithChore);
    return houseWithChore;
  case 'ADD_READER':
    const oldBulletin = action.usersHouse.bulletins.find(bulletin => bulletin.id === action.bulletinId);
    if (oldBulletin.hasRead.includes(action.userId)) {
      return action.usersHouse;
    }
    const newBulletin = Object.assign(oldBulletin, { hasRead: [...oldBulletin.hasRead, action.userId] });
    const updateBulletins = action.usersHouse.bulletins.filter(bulletin => {
      return bulletin.id !== action.bulletinId;
    });
    const updateWithBulletin = Object.assign(action.usersHouse, { bulletins: [...updateBulletins, newBulletin] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updateWithBulletin);
    return updateWithBulletin;
  case 'MARK_PAID':
    const oldBill = action.usersHouse.bills.find(bill => bill.id === action.billId);
    const user = oldBill.allUsersTotals.find(user => user.id === action.userId);
    if (user.paid === true) {
      return action.usersHouse;
    }
    const newUser = Object.assign(user, { paid: true });
    const newAllUsersTotals = oldBill.allUsersTotals.filter(user => user.id !== action.userId);
    const newBill = Object.assign(oldBill, { allUsersTotals: [...newAllUsersTotals, newUser] });
    const updatedBills = action.usersHouse.bills.filter(bill => {
      return bill.id !== action.billId;
    });
    const updateWithBill = Object.assign(action.usersHouse, { bills: [...updatedBills, newBill] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updateWithBill);
    return updateWithBill;
  default:
    return state;
  }
};

export default usersHouse;
