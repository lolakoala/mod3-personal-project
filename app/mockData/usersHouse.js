import currentUser1 from './currentUser1.js';
import currentUser2 from './currentUser2.js';

const bill1 = {
  allUsersTotals: [
    {
      id: currentUser1.id,
      paid: false,
      total: 50
    },
    {
      id: currentUser2.id,
      paid: false,
      total: 50
    }
  ],
  datePosted: "11/01/2017",
  details: "We used a lot of water.",
  duedate: "11/9/17",
  id: "b3019762705779",
  parsedDuedate: 1510210800000,
  postedBy: currentUser1,
  title: "Water Bill",
  total: "100"
};

const bulletin1 = {
  datePosted: "11/01/2017",
  details: "The dogs got out through a hole in the fence yesterday.",
  hasRead: [currentUser1.id],
  id: 1509567098747,
  postedBy: currentUser1,
  title: "Don't leave dogs out unattended."
};

const chore1 = {
  assignedTo: currentUser1.id,
  datePosted: "10/31/2017",
  details: "Empty the dishwasher.",
  done: false,
  id: "c1509485241076",
  postedBy: currentUser1,
  title: "Dishes",
  urgency: "low"
};

const usersHouse = {
  houseKey: "-KxoIBWy5UYRscFYdWty",
  bills: [bill1],
  bulletins: [bulletin1],
  chores: [chore1],
  houseCode: "woof9",
  houseName: "Dog9",
  users: [currentUser1, currentUser2]
};

export default usersHouse;
