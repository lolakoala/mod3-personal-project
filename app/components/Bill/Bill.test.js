import React from "react";
import Bill from './Bill.js';
import { shallow } from "enzyme";
import usersHouse from '../../mockData/usersHouse.js';
import currentUser1 from '../../mockData/currentUser1';

describe('Bill component', () => {
  const mockFn = jest.fn();
  const bill = usersHouse.bills[0];
  const match = {
    params: {
      id: "b3019762705779"
    }
  };
  const wrapper = shallow(<Bill
    usersHouse={usersHouse}
    markBillPaid={mockFn}
    currentUser={currentUser1}
    match={match}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a title', () => {
    const title = wrapper.find('.title');

    expect(title.text()).toEqual(`Bill: ${bill.title}`);
  });

  it('should have a dutedate', () => {
    const duedate = wrapper.find('.duedate');

    expect(duedate.text()).toEqual(`Due: ${bill.duedate}`);
  });

  it('should have a total', () => {
    const total = wrapper.find('.total');

    expect(total.text()).toEqual(`Total: ${bill.total}`);
  });

  it('should have posted info', () => {
    const postedInfo = wrapper.find('.posted-by-on');

    expect(postedInfo.text()).toEqual(`Posted by ${bill.postedBy.name} on ${bill.datePosted}`);
  });

  it('should have details', () => {
    const details = wrapper.find('.details');

    expect(details.text()).toEqual(`Details: ${bill.details}`);
  });

  it('should have a split total for each house member', () => {
    const splits = wrapper.find('.user-split');

    expect(splits.length).toEqual(bill.allUsersTotals.length);
  });

  it('should fire function', () => {
    const markPaid = wrapper.find('.mark-paid').first();

    markPaid.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
