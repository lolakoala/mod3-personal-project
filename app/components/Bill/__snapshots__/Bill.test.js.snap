// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bill
    currentUser={
      Object {
        "house": "Dog9",
        "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
        "name": "Amanda Brenner",
      }
    }
    markBillPaid={[Function]}
    match={
      Object {
        "params": Object {
          "id": "b3019762705779",
        },
      }
    }
    usersHouse={
      Object {
        "bills": Array [
          Object {
            "allUsersTotals": Array [
              Object {
                "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "paid": false,
                "total": 50,
              },
              Object {
                "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                "paid": false,
                "total": 50,
              },
            ],
            "datePosted": "11/01/2017",
            "details": "We used a lot of water.",
            "duedate": "11/9/17",
            "id": "b3019762705779",
            "parsedDuedate": 1510210800000,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Water Bill",
            "total": "100",
          },
        ],
        "bulletins": Array [
          Object {
            "datePosted": "11/01/2017",
            "details": "The dogs got out through a hole in the fence yesterday.",
            "hasRead": Array [
              "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            ],
            "id": 1509567098747,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Don't leave dogs out unattended.",
          },
        ],
        "chores": Array [
          Object {
            "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "datePosted": "10/31/2017",
            "details": "Empty the dishwasher.",
            "done": false,
            "id": "c1509485241076",
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Dishes",
            "urgency": "low",
          },
        ],
        "houseCode": "woof9",
        "houseKey": "-KxoIBWy5UYRscFYdWty",
        "houseName": "Dog9",
        "users": Array [
          Object {
            "house": "Dog9",
            "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "name": "Amanda Brenner",
          },
          Object {
            "house": "Dog9",
            "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
            "name": "Lola Koala",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="title"
        >
          Bill: Water Bill
        </p>,
        <div
          className="due-total"
        >
          <p
            className="duedate"
          >
            Due: 11/9/17
          </p>
          <p
            className="total"
          >
            Total: 100
          </p>
        </div>,
        <p
          className="details"
        >
          Details: We used a lot of water.
        </p>,
        <p
          className="split"
        >
          House Split...
        </p>,
        Array [
          <div
            className="user-split"
          >
            <p>
              Amanda Brenner: 50
            </p>
            <div
              className="mark-paid"
              onClick={[Function]}
            >
              <p>
                Mark as Paid
              </p>
            </div>
          </div>,
          <div
            className="user-split"
          >
            <p>
              Lola Koala: 50
            </p>
            <div
              className="mark-paid"
              onClick={[Function]}
            >
              <p>
                Not Yet Paid
              </p>
            </div>
          </div>,
        ],
        <p
          className="posted-by-on"
        >
          Posted by Amanda Brenner on 11/01/2017
        </p>,
      ],
      "id": "bill-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Bill: Water Bill",
          "className": "title",
        },
        "ref": null,
        "rendered": "Bill: Water Bill",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="duedate"
            >
              Due: 11/9/17
            </p>,
            <p
              className="total"
            >
              Total: 100
            </p>,
          ],
          "className": "due-total",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Due: 11/9/17",
              "className": "duedate",
            },
            "ref": null,
            "rendered": "Due: 11/9/17",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Total: 100",
              "className": "total",
            },
            "ref": null,
            "rendered": "Total: 100",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Details: We used a lot of water.",
          "className": "details",
        },
        "ref": null,
        "rendered": "Details: We used a lot of water.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "House Split...",
          "className": "split",
        },
        "ref": null,
        "rendered": "House Split...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1510210800000 RVde1m9BhkQBLZlTJNsFlQZXqP43",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Amanda Brenner: 50
            </p>,
            <div
              className="mark-paid"
              onClick={[Function]}
            >
              <p>
                Mark as Paid
              </p>
            </div>,
          ],
          "className": "user-split",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Amanda Brenner: 50",
            },
            "ref": null,
            "rendered": "Amanda Brenner: 50",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Mark as Paid
              </p>,
              "className": "mark-paid",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Mark as Paid",
              },
              "ref": null,
              "rendered": "Mark as Paid",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1510210800000 cFNdorl5ceRyMr9XGvPFRVEcw7I3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Lola Koala: 50
            </p>,
            <div
              className="mark-paid"
              onClick={[Function]}
            >
              <p>
                Not Yet Paid
              </p>
            </div>,
          ],
          "className": "user-split",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Lola Koala: 50",
            },
            "ref": null,
            "rendered": "Lola Koala: 50",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Not Yet Paid
              </p>,
              "className": "mark-paid",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Not Yet Paid",
              },
              "ref": null,
              "rendered": "Not Yet Paid",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Posted by Amanda Brenner on 11/01/2017",
          "className": "posted-by-on",
        },
        "ref": null,
        "rendered": "Posted by Amanda Brenner on 11/01/2017",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="title"
          >
            Bill: Water Bill
          </p>,
          <div
            className="due-total"
          >
            <p
              className="duedate"
            >
              Due: 11/9/17
            </p>
            <p
              className="total"
            >
              Total: 100
            </p>
          </div>,
          <p
            className="details"
          >
            Details: We used a lot of water.
          </p>,
          <p
            className="split"
          >
            House Split...
          </p>,
          Array [
            <div
              className="user-split"
            >
              <p>
                Amanda Brenner: 50
              </p>
              <div
                className="mark-paid"
                onClick={[Function]}
              >
                <p>
                  Mark as Paid
                </p>
              </div>
            </div>,
            <div
              className="user-split"
            >
              <p>
                Lola Koala: 50
              </p>
              <div
                className="mark-paid"
                onClick={[Function]}
              >
                <p>
                  Not Yet Paid
                </p>
              </div>
            </div>,
          ],
          <p
            className="posted-by-on"
          >
            Posted by Amanda Brenner on 11/01/2017
          </p>,
        ],
        "id": "bill-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Bill: Water Bill",
            "className": "title",
          },
          "ref": null,
          "rendered": "Bill: Water Bill",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="duedate"
              >
                Due: 11/9/17
              </p>,
              <p
                className="total"
              >
                Total: 100
              </p>,
            ],
            "className": "due-total",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Due: 11/9/17",
                "className": "duedate",
              },
              "ref": null,
              "rendered": "Due: 11/9/17",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Total: 100",
                "className": "total",
              },
              "ref": null,
              "rendered": "Total: 100",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Details: We used a lot of water.",
            "className": "details",
          },
          "ref": null,
          "rendered": "Details: We used a lot of water.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "House Split...",
            "className": "split",
          },
          "ref": null,
          "rendered": "House Split...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1510210800000 RVde1m9BhkQBLZlTJNsFlQZXqP43",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Amanda Brenner: 50
              </p>,
              <div
                className="mark-paid"
                onClick={[Function]}
              >
                <p>
                  Mark as Paid
                </p>
              </div>,
            ],
            "className": "user-split",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Amanda Brenner: 50",
              },
              "ref": null,
              "rendered": "Amanda Brenner: 50",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Mark as Paid
                </p>,
                "className": "mark-paid",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Mark as Paid",
                },
                "ref": null,
                "rendered": "Mark as Paid",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1510210800000 cFNdorl5ceRyMr9XGvPFRVEcw7I3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Lola Koala: 50
              </p>,
              <div
                className="mark-paid"
                onClick={[Function]}
              >
                <p>
                  Not Yet Paid
                </p>
              </div>,
            ],
            "className": "user-split",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Lola Koala: 50",
              },
              "ref": null,
              "rendered": "Lola Koala: 50",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Not Yet Paid
                </p>,
                "className": "mark-paid",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Not Yet Paid",
                },
                "ref": null,
                "rendered": "Not Yet Paid",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Posted by Amanda Brenner on 11/01/2017",
            "className": "posted-by-on",
          },
          "ref": null,
          "rendered": "Posted by Amanda Brenner on 11/01/2017",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
